### DISCUSSION TITLE
Resolved: Bring `rust-bitcoind-json-rpc` into the `rust-bitcoin` org

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3225] [General]
I would like to request that the [`rust-bitcoind-json-rpc`](https://github.com/tcharding/rust-bitcoind-json-rpc) repository be moved into the `rust-bitcoin` org. It is now used by `rust-miniscript` and meets the requirements of:

- Has CI
- Adheres to `rust-bitcoin` org coding style and standards

(I can't remember where we documented what our requirements are to be part of the org.)

EDIT: The requirements are documented in the org readme: https://github.com/rust-bitcoin

**Comment by apoelstra** [comment]
Let's do it.

**Reply by apoelstra** [reply]
I believe you need to take action since it's currently under your personal org.

**Reply by tcharding** [reply]
Ah yes, thanks. Done!

**Comment by storopoli** [comment]
+1 for this. I'm gonna use this in production.

**Reply by tcharding** [reply]
I'm reading this as a joke :)  (and commenting just so someone else does not miss that it is :)

**Reply by tnull** [reply]
Is it though? I'm going to start a project requiring support for `submitpackage` real soon, and at this point rather not bother pulling in `rust-bitcoincore-rpc`. So it's either completely rolling my own, or using/forking said 'non-production' code where at least other people already had an eye on. 

@tcharding Would you accept a pull request adding `submitpackage`?

**Reply by tcharding** [reply]
Most definitely! The plan is to have `rust-bitcoind-json`  test every JSONRPC method provided by Core from 17.1 onwards. And hence `bitcoind-json-rpc-types` should then provide the types any project needs. The "don't use in production" bit refers to `bitcoind-json-rpc-client` - I'm hoping folk will write their own clients using `bitcoind-json-rpc-types` (and copying code from `client` if they want to). In otherwords, your production client should only support the RPC methods you use, and then you implement the features that are important to your project instead of raising issues on the communal repo. That is the dream anyway, there is a bunch of grind work to do now to get us there.

**Reply by tnull** [reply]
> Most definitely! 

Great, I intend to start working on it ~end of this week.

> I'm hoping folk will write their own clients using bitcoind-json-rpc-types (and copying code from client if they want to). In otherwords, your production client should only support the RPC methods you use, and then you implement the features that are important to your project instead of raising issues on the communal repo. 

Tbh., I'm not fully sold on this approach, as a production-ready `Client` would likely be required almost everywhere you'd need to deal with Core's RPC? So if `rust-bitocoincore-rpc` is more or less deprecated, having a reusable client would IMO be beneficial for the ecosystem. I understand that there might be concerns around additional maintenance burden, but possibly this could be somewhat mitigated by bringing on some additional maintainers for the crate? Are there any other notable concerns?

**Reply by tcharding** [reply]
No concerns other than I have basically zero interest in all things webby. More than happy to have other maintainers/owners of the crates.

**Reply by storopoli** [reply]
the "reference" client in `bitcoind-json-rpc-client` uses `minreq` as a client, what would you use for a "reference" `async` client?

**Comment by Kixunil** [comment]
Potentially relevant: https://github.com/rust-bitcoin/rust-bitcoin/issues/3239

**Comment by tcharding** [comment]
This is marked resolved because the code lives in https://github.com/rust-bitcoin/corepc now.

**Comment by tcharding** [comment]
I cannot remember why I started manually marking discussions as 'Resolved: ...' instead of just closing them.
